music_assistant/providers/webdav/__init__.py

```diff
@@ -0,0 +1,108 @@
+"""WebDAV filesystem provider for Music Assistant."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from music_assistant_models.config_entries import ConfigEntry, ConfigValueOption
+from music_assistant_models.enums import ConfigEntryType, ProviderFeature
+
+from music_assistant.constants import CONF_PASSWORD, CONF_USERNAME
+
+from .constants import CONF_CONTENT_TYPE, CONF_URL, CONF_VERIFY_SSL
+from .provider import WebDAVFileSystemProvider
+
+if TYPE_CHECKING:
+    from music_assistant_models.config_entries import ConfigValueType, ProviderConfig
+    from music_assistant_models.provider import ProviderManifest
+
+    from music_assistant.mass import MusicAssistant
+    from music_assistant.models import ProviderInstanceType
+
+
+# Supported features
+SUPPORTED_FEATURES = {
+    ProviderFeature.BROWSE,
+    ProviderFeature.SEARCH,
+}
+
+
+async def setup(
+    mass: MusicAssistant, manifest: ProviderManifest, config: ProviderConfig
+) -> ProviderInstanceType:
+    """Initialize provider(instance) with given configuration."""
+    prov = WebDAVFileSystemProvider(mass, manifest, config)
+    await prov.handle_async_init()
+    return prov
+
+
+async def get_config_entries(
+    mass: MusicAssistant,
+    instance_id: str | None = None,
+    action: str | None = None,
+    values: dict[str, ConfigValueType] | None = None,
+) -> tuple[ConfigEntry, ...]:
+    """Return Config entries to setup this provider."""
+    # ruff: noqa: ARG001
+    return (
+        ConfigEntry(
+            key=CONF_CONTENT_TYPE,
+            type=ConfigEntryType.STRING,
+            label="Content type",
+            options=[
+                ConfigValueOption("Music", "music"),
+                ConfigValueOption("Audiobooks", "audiobooks"),
+                ConfigValueOption("Podcasts", "podcasts"),
+            ],
+            default_value="music",
+            description="The type of content stored on this WebDAV server",
+            hidden=instance_id is not None,
+        ),
```

Comment:
Cant you import this from filesystem prov constants ?

Reply:
fixed